src/solver/simulation/include/antares/solver/simulation/remix-storage/storage-for-remix.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "storage-for-remix-interface.h"
+
+namespace Antares::Solver::Simulation
+{
+
+class StorageForRemixNoLevels: public IStorageForRemix
+{
+public:
+    StorageForRemixNoLevels(std::vector<double>& generation,
+                            std::vector<double>& unsupE,
+                            const std::vector<double> Pmax,
+                            const std::vector<double> Pmin);
+
+    double maxExchange(unsigned hourOfMaxGen, unsigned hourOfMinGen) const override;
+    void update() override;
+    const std::vector<double>& initWithdrawal() override;
+    std::vector<double>& withdrawal() override;
+
+protected:
+    void checkInput(size_t size) override;
+
+    std::vector<double>& withdrawal_;
+    const std::vector<double> initWithdrawal_;
+    std::vector<double>& unsupE_;
+    const std::vector<double> pmax_;
+    const std::vector<double> pmin_;
+};
+
+class StorageForRemixWithLevels: public StorageForRemixNoLevels
+{
+public:
+    StorageForRemixWithLevels(std::vector<double>& withdrawal,
+                              std::vector<double>& unsupE,
+                              std::vector<double>& levels,
+                              const std::vector<double> Pmax,
+                              const std::vector<double> Pmin,
+                              const std::vector<double> inflows,
+                              const std::vector<double> overflow,
+                              const std::vector<double>& injection,
+                              const std::vector<double> lowRuleCurve,
+                              const std::vector<double> upRuleCurve,
+                              const double initLevel,
+                              const double withdrawalEff,
+                              const double injectionEff);
```

Comment:
I see that the caller must provide a `withdrawalEff` & `injectionEff`, while there are also
```cpp
    const double withdrawalEff_ = 1;
    const double injectionEff_ = 1;
```
right below. Why ?

Reply:
These 2 have a default value now